dotnet/Gherkin/Parser.cs

```diff
@@ -89,7 +101,12 @@ protected class ParserContext
 
         public T Parse(ITokenScanner tokenScanner)
         {
-            return Parse(tokenScanner, new TokenMatcher(), new AstBuilder<T>());
+            return Parse(tokenScanner, new TokenMatcher(), this.AstBuilder);
+        }
+
+        public T Parse(TokenScanner tokenScanner, TokenMatcher tokenMatcher)
+        {
+            return Parse(tokenScanner, tokenMatcher, this.AstBuilder);

```

Comment:
Don't pass `this.AstBuilder` around. It's an instance variable. Also, remove it from the next method and from `ParserContext`.